h2o-r/h2o-package/R/connection.R

```diff
@@ -758,7 +742,10 @@ h2o.resume <- function(recovery_dir=NULL) {
 # or it calls stop() and does not return.
 #
 # It will download a jar file if it needs to.
+#' @importFrom utils flush.console
 .h2o.downloadJar <- function(overwrite = FALSE) {
+  old_options <- options(timeout = max(1000, getOption("timeout")))
+  on.exit(options(old_options))
```

Comment:
@tomasfryda  Can you explain what this does?  To me, it looks like it will save the "new" timeout option of 1000s (but you call it `old_options`).  And then it sets the timeout to 1000s only when the `.h2o.downloadJar()` exits (which would be after the download instead of before it).  It seems like it does the opposite of what we want it to do?  I might be missing something though.

Would we do something like this instead?
```
old_options <- options(timeout = getOption("timeout")) #save what the user had
options(timeout = max(1000, getOption("timeout"))) #set the 1000s for the download
on.exit(options(old_options)) #return to the previous settings
  ```

Reply:
cc @jangorecki Can you confirm?

Reply:
@ledell From the options documentation(https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/options)

>For options(), a list of all set options sorted by name. For options(name), a list of length one containing the set value, or NULL if it is unset. **For uses setting one or more options, a list with the previous values of the options changed (returned invisibly).**

Reply:
@tomasfryda Ok, that's not what I expected!  Thanks for the reference.  

We can ignore my comments above -- the code does what we want it to.  So I think we are good to merge!